src/ec2/model.ts

```diff
@@ -113,19 +115,29 @@ export class Ec2ConnectionManager {
         const hasPermission = await this.hasProperPermissions(IamRole!.Arn)
 
         if (!hasPermission) {
-            const message = `Ensure an IAM role with the proper permissions is attached to the instance. Found attached role: ${
-                IamRole!.Arn
-            }`
-            this.throwConnectionError(message, selection, {
-                code: 'EC2SSMPermission',
-                documentationUri: this.policyDocumentationUri,
-            })
+            const policiesAdded = await promptToAddPolicies(this.iamClient, IamRole!.Arn!)
+            if (!policiesAdded) {
+                const message = `Did not add permissions. Ensure an IAM role with the proper permissions is attached to the instance. Found attached role: ${
+                    IamRole!.Arn
+                }`
+                this.throwConnectionError(message, selection, {
+                    code: 'EC2SSMPermission',
+                    documentationUri: this.policyDocumentationUri,
+                })
+            } else {
+                const timeout = new Timeout(policyAttachDelay)
```

Comment:
Is it cheap to poll an api until we can confirm the policy is attached?

Reply:
I tried a polling approach on if the policy is attached, but it appears the SDK call will tell me its attached, before it actually works. When I stepped through with debugger, the polling would always succeed on the first attempt, then open terminal and it still crashes.

Reply:
It's possible I had polling implemented incorrectly, so I committed the function I used in the recent commit.